.github/workflows/storybook-chromatic.yml

```
@@ -263,19 +291,20 @@ jobs:
                   pnpm --filter=@posthog/storybook test:visual:ci:$VARIANT --browsers ${{ matrix.browser }} --shard ${{ matrix.shard }}/${{ matrix.shard_count }}
 
             - name: Archive failure screenshots
-              if: ${{ failure() }}
+              if: needs.changes.outputs.frontend == 'true' && ${{ failure() }}
```

actionlint (1.7.7)

if: condition "needs.changes.outputs.frontend == 'true' && ${{ failure() }}" is always evaluated to true because extra characters are around ${{ }}

(if-cond)